File: python/get_options.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
options_api = intrinio_sdk.OptionsApi()
 
symbol = 'AAPL' # str | The option symbol, corresponding to the underlying security.
type = 'put' # str | The option contract type. (optional)

try:
    api_response = options_api.get_options(symbol, type=type)
    options = api_response.options

    print("%s options found for %s!" % (len(options), symbol))

    for option in options:
        print("")
        print("ID:         %s" % (option.id))
        print("Code:       %s" % (option.code))
        print("Ticker:     %s" % (option.ticker))
        print("Expiration: %s" % (option.expiration))
        print("Strike:     %s" % (option.strike))
        print("Type:       %s" % (option.type))
except ApiException as e:
    print("Exception when calling OptionsApi->get_options: %s\n" % e)

File: csharp/GetOptions.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
  public class GetOptionsExample
  {
    public static void Main()
    {
      Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

      var optionsApi = new OptionsApi();
      string symbol = "AAPL";  // string | The option symbol, corresponding to the underlying security.
      string type = "put";  // string | The option contract type. (optional) 
      decimal? strike = null;  // decimal? | The strike price of the option contract. This will return options contracts with strike price equal to this price. (optional) 
      decimal? strikeGreaterThan = null;  // decimal? | The strike price of the option contract. This will return options contracts with strike prices greater than this price. (optional) 
      decimal? strikeLessThan = null;  // decimal? | The strike price of the option contract. This will return options contracts with strike prices less than this price. (optional) 
      string expiration = null;  // string | The expiration date of the option contract. This will return options contracts with expiration dates on this date. (optional) 
      string expirationAfter = null;  // string | The expiration date of the option contract. This will return options contracts with expiration dates after this date. (optional) 
      string expirationBefore = null;  // string | The expiration date of the option contract. This will return options contracts with expiration dates before this date. (optional) 
      decimal? pageSize = null;  // decimal? | The number of results to return (optional)  (default to 100)
      string nextPage = null;  // string | Gets the next page of data from a previous API call (optional) 

      try
      {
        ApiResponseOptions result = optionsApi.GetOptions(symbol, type, strike, strikeGreaterThan, strikeLessThan, expiration, expirationAfter, expirationBefore, pageSize, nextPage);
        List<Option> options = result.Options;

        Console.WriteLine(options.Count + " options found for " + symbol + "!");

        options.ForEach(delegate (Option option)
        {
          Console.WriteLine();
          Console.WriteLine("ID:         " + option.Id);
          Console.WriteLine("Code:       " + option.Code);
          Console.WriteLine("Ticker:     " + option.Ticker);
          Console.WriteLine("Expiration: " + option.Expiration);
          Console.WriteLine("Strike:     " + option.Strike);
          Console.WriteLine("Type:       " + option.Type);
        });
      }
      catch (Exception e)
      {
        Debug.Print("Exception when calling OptionsApi.GetOptions: " + e.Message);
      }
    }
  }
}
